backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/NumaPolicyUnit.java

```diff
@@ -57,6 +57,14 @@ public NumaPolicyUnit(PolicyUnit policyUnit,
 
         List<VDS> result = new ArrayList<>();
         for (VDS host: hosts) {
+            if (vmsToCheck.stream().anyMatch(VmCpuCountHelper::isAutoPinning) &&
+                    host.getCpuSockets() != host.getNumaNodeList().size()) {
```

Comment:
Requested to check before iterating over vmsToCheck and skip VMs that already run on the host.
Need to check that the filtering works on initial start.